pkg/cmd/gist/list/http.go

```diff
@@ -0,0 +1,45 @@
+package list
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/pkg/cmd/gist/shared"
+)
+
+func listGists(client *http.Client, hostname string, limit int, visibility string) ([]shared.Gist, error) {
+	result := []shared.Gist{}
+
+	query := url.Values{}
+	if visibility == "all" {
+		query.Add("per_page", fmt.Sprintf("%d", limit))
+	} else {
+		query.Add("per_page", "100")
+	}
+
+	apiClient := api.NewClientFromHTTP(client)
+	err := apiClient.REST(hostname, "GET", "gists?"+query.Encode(), nil, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	gists := []shared.Gist{}
+
+	for _, gist := range result {
+		if len(gists) == limit {
+			break
+		}
+		if visibility == "all" || (visibility == "secret" && !gist.Public) || (visibility == "public" && gist.Public) {
```

Comment:
With GraphQL, I think you might be able to fetch gists based on their private bit and order them by UPDATED_AT.

Reply:
ah interesting, for some reason my brain thought "gists are not in the graphql api" but there they are.

i'm going to finish writing tests and then if i have time circle back and port this to graphql.